src/rgw/rgw_datalog.h

```diff
@@ -50,23 +50,29 @@ struct rgw_data_change {
   DataLogEntityType entity_type;
   std::string key;
   ceph::real_time timestamp;
+  uint64_t gen_id;
 
   void encode(ceph::buffer::list& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 2, bl);
```

Comment:
this raises the `compat` version to 2, which means that `DECODE_START(2, bl)` below would throw if it gets a `struct_v < 2`, and we wouldn't reach this backward-compatible decode logic:
```c++
     if (struct_v < 2)
       gen_id = 0;
     else
       decode(gen_id, bl);
```
this would cause issues with rolling upgrades in a zone with multiple radosgws